packages/core/type-definitions/combinator/merge.d.ts

```diff
@@ -2,4 +2,8 @@ import { Stream } from '@most/types';
 
 export function merge<A, B>(s1: Stream<A>, s2: Stream<B>): Stream<A | B>;
 export function merge<A, B>(s1: Stream<A>): (s2: Stream<B>) => Stream<A | B>
-export function mergeArray<A>(streams: Array<Stream<A>>): Stream<A>;
+
+type MergeArray<S extends ReadonlyArray<Stream<any>>> = Value<S[number]>
+type Value<S> = S extends Stream<infer A> ? A : never
+
+export function mergeArray<S extends ReadonlyArray<Stream<any>>>(streams: S): Stream<MergeArray<S>>;
```

Comment:
Consider `readonly Stream<any>[]`

Reply:
Ah, I do like that.  My only concern is compatibility.  Do you know in which TS version that syntax was introduced?  If it's relatively new, we may want to wait a minor version or two, but if it's already been around a while, we can probably go with it.

Reply:
Looks like it was introduced [in 3.4.0](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html).  Thoughts on waiting until 3.5.0 is released vs using `readonly` now while 3.4.x is the latest minor?

Reply:
Since it was released with v3.4.0, a non-breaking feature release, I don’t see any reason to delay the use. I mean, what would be the point of waiting for v3.5.0? If we delay the application, then, as a minimum, I think we should add TODO comments.

Reply:
I'm thinking more in terms of breaking compatibility with anyone using TS < 3.4.  We'd be introducing a type that breaks their build.  This seems like a slightly gray area for semver, I guess strictly speaking, we'd need to release a new major version of `@most/core`.

I like the TODO idea, or even create a github issue now to switch from `ReadonlyArray` to `readonly`.

Reply:
I think I'm in favor of adding TODO comments and waiting a few releases. Virtually *every* TypeScript release is a breaking change and updating to 3.4 syntax will break any users not yet on that version.

Reply:
> Virtually every TypeScript release is a breaking change

😬 yikes!

Ok, sounds like we're all ok with adding TODOs for now.  I'll push an update with those soon, and then think about what we should do about documenting this in the face of the TS/Flow disparity.

Reply:
Yeah to clarify my previous statement - every `minor` release is potentially breaking. Every patch release in theory shouldn't be. I don't believe they follow semver in any regards TBH. I had a lot more run ins with that during the 2.x releases but I still don't really trust the version numbers